File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Raindrops
  def self.convert(input)
    output =
        [input % 3 == 0 ? "Pling" : '',
         input % 5 == 0 ? "Plang" : '',
         input % 7 == 0 ? "Plong" : ''
        ].join("")

    output.empty? ? input.to_s : output
  end
end

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
import org.jetbrains.annotations.NotNull;

object Raindrops {

    @NotNull
    fun convert(input: Int): String {
        var result: String = listOf(
                if (input % 3 == 0) "Pling" else "",
                if (input % 5 == 0) "Plang" else "",
                if (input % 7 == 0) "Plong" else ""
        ).joinToString(separator = "")
        return if (result.equals("")) "${input}" else result
    }
}
